### PR TITLE
Asychronous observable service + message service

### PR DESCRIPTION
**Pull request by athompsonScottLogic**

### CODE REVIEW

File: src/app/heroes/heroes.component.ts [hunk 1]
```diff
@@ -9,22 +10,23 @@ import { HEROES } from '../mock-heroes';
...
-    name: 'Windstorm'
-  };
-
-  constructor() { }
-
-  ngOnInit(): void {
+  constructor(private heroService: HeroService, private messageService: MessageService) {}
+  
+  getHeroes(): void {
+    this.heroService.getHeroes()
```
Comment (connor-settle) [inline src/app/heroes/heroes.component.ts]: avoid explicit subscribe. Look into https://angular.io/api/common/AsyncPipe 

It makes the template handle subscriptions (and importantly unsubscribe).

https://paulrohan.medium.com/angular-avoiding-subscribe-method-by-replacing-it-with-an-asynpipe-when-possible-a92c20793357

File: src/app/message.service.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import { Injectable } from '@angular/core';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class MessageService {
```
Comment (connor-settle) [inline src/app/message.service.ts:6]: This is starting to look like a store :) 

I guess a good candidate to move to ngrx when you get to that part.

File: src/app/messages/messages.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+  it('should display message title', () => {
+    fixture.detectChanges();
+    expect(h2.textContent).toContain("Messages");
+  });
+
+  it('should display a different message title', () => {
+    h2.textContent = 'Oodles of Noodles';
+    fixture.detectChanges();
+    expect(h2.textContent).toContain('Oodles of Noodles');
```
Comment (connor-settle) [inline src/app/messages/messages.component.spec.ts]: what are you testing here? Seems to me that you are testing setting textContent works as expected - which is not your code.

**Review by connor-settle — CHANGES_REQUESTED** [review]

File: src/app/messages/messages.component.spec.ts (hunk 3 above)
Comment (athompsonScottLogic) [reply src/app/messages/messages.component.spec.ts]: Ah, whoops. I was just learning the syntax. Didn't mean to leave that in.

File: src/app/heroes/heroes.component.ts (hunk 1 above)
Comment (athompsonScottLogic) [reply src/app/heroes/heroes.component.ts]: Just converted that function to use asyncpipe via the use of an observable. Does that look right?

File: src/app/heroes/heroes.component.ts [hunk 4]
```diff
@@ -10,14 +11,13 @@ import { MessageService } from '../message.service';
 })
 export class HeroesComponent implements OnInit {
 
-  heroes: Hero[] = [];
+  public heroes$!: Observable<Hero[]>;
```
Comment (connor-settle) [inline src/app/heroes/heroes.component.ts]: Access modifier is not required. Its public by default.

You might want to consider just initializing it when the component is created, i.e. 
```ts
heroes$ = this.heroService.getHeroes();
```

File: src/app/heroes/heroes.component.ts (hunk 1 above)
Comment (connor-settle) [reply src/app/heroes/heroes.component.ts]: async pipe usage looks good. I've added another comment for an improvement

File: src/app/heroes/heroes.component.ts (hunk 4 above)
Comment (athompsonScottLogic) [reply src/app/heroes/heroes.component.ts]: Done

**Review by connor-settle — APPROVED** [review]

**Pull request merged**